### PR TITLE
Add options "undef", "replace" form CSV#open

### PR DESCRIPTION
**Pull request by yuuna**
I want to use CSV#open with undef: :repace, replace: "*" options form wrong character.

### CODE REVIEW

File: lib/csv.rb [hunk 1]
```diff
@@ -936,7 +943,9 @@ def initialize(data,
                  write_converters: nil,
                  write_nil_value: nil,
                  write_empty_value: "",
-                 strip: false)
+                 strip: false,
+                 undef: nil,
+                 replace: nil)
```
Comment (kou) [inline lib/csv.rb]: Should we need to accept `:undef` and `:replace` in `CVS#initialize`?
It seems that we just need to accept them in `CSV.open`. Then we can remove them from `options` in `CSV.open` arguments.

File: test/csv/interface/test_read.rb [hunk 2]
```diff
@@ -264,6 +264,14 @@ def test_new_nil
     end
   end
 
+  def test_foreach_undef_replace
+    rows = []
+    CSV.foreach(@input.path, col_sep: "\t", row_sep: "\r\n", undef: :replace, replace: "*").each do |row|
+      rows << row
+    end
+    assert_equal(@rows, rows)
+  end
```
Comment (kou) [inline test/csv/interface/test_read.rb]: We need to new test data that fails without these new options.

File: lib/csv.rb (hunk 1 above)
Comment (yuuna) [reply lib/csv.rb]: I will fix  the arguments in CSV.open.

File: test/csv/interface/test_read.rb (hunk 2 above)
Comment (yuuna) [reply test/csv/interface/test_read.rb]: I will make a new test data.

**Comment by kou** [comment]
Orphan.

**Pull request closed without merging**